pkg/controller/issueapi/issue.go

```diff
@@ -177,10 +188,10 @@ func (c *Controller) recordStats(ctx context.Context, results []*IssueResult) {
 
 // smsProviderFor returns the sms provider for the given realm. It pulls the
 // value from a local in-memory cache.
-func (c *Controller) smsProviderFor(ctx context.Context, realm *database.Realm) (sms.Provider, error) {
-	key := fmt.Sprintf("realm:%d:sms_provider", realm.ID)
+func (c *Controller) smsProviderFor(ctx context.Context, realm *database.Realm, userReport bool) (sms.Provider, error) {
+	key := fmt.Sprintf("realm:%d:sms_provider:user_report:%v", realm.ID, userReport)
```

Comment:
One of the advantages of making user report a separate sms provider is that we can keep this logic the same everywhere, since it'd have a unique ID.

Reply:
I think we agreed not to do that, right?

I've made this contingent on the options applied and the order they appear in - best future proofing I can think of ATM.

Reply:
Yea, agreed.